test/server/cards/02_SHD/units/DoctorPershingExperimentingWithLife.spec.ts

```diff
@@ -0,0 +1,86 @@
+describe('Doctor Pershing, Experimenting With Life', function() {
+    integration(function(contextRef) {
+        describe('Doctor Pershing\'s action ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['doctor-pershing#experimenting-with-life', 'wampa'],
+                        spaceArena: ['tieln-fighter']
+                    },
+                    player2: {
+                        groundArena: ['frontier-atrt'],
+                        spaceArena: ['cartel-spacer']
+                    }
+                });
+            });
+
+            it('should deal 1 damage to any selected friendly unit and draw a card', function () {
+                const { context } = contextRef;
+
+                const reset = () => {
+                    context.player2.passAction();
+                    context.doctorPershing.exhausted = false;
+                };
+
+                // Trigger the ability
+                context.player1.clickCard(context.doctorPershing);
+                context.player1.clickPrompt('Draw a card');
+                expect(context.player1).toBeAbleToSelectExactly([context.doctorPershing, context.wampa, context.tielnFighter]);
+                expect(context.player1).not.toHaveChooseNoTargetButton();
+
+                // Target a space friendly unit
+                context.player1.clickCard(context.tielnFighter);
+                expect(context.doctorPershing.exhausted).toBeTrue();
+                expect(context.tielnFighter).toBeInLocation('discard');
+                expect(context.player1.hand.length).toBe(1);
+                expect(context.player2.hand.length).toBe(0);
+
+                // Trigger the ability again
+                reset();
+                context.player1.clickCard(context.doctorPershing);
+                context.player1.clickPrompt('Draw a card');
+                expect(context.player1).toBeAbleToSelectExactly([context.doctorPershing, context.wampa]);
+                expect(context.player1).not.toHaveChooseNoTargetButton();
+
+                // Target a ground friendly unit
+                context.player1.clickCard(context.wampa);
```

Comment:
No need to make changes here but typically it's safe to assume for testing purposes that any unit which appears as selectable will work correctly and don't need to test all cases. Only exceptions are when there might be distinct behavior differences depending on the target (e.g. targeting self is usually a worthwhile extra case), but if it's just e.g. arena or controller differences it's usually fine. Really appreciate the thorough tests.

Reply:
Makes sense! I gave the Wampa a shield now.